ocaml/parsing/ast_helper.mli

```diff
@@ -190,18 +190,10 @@ module Exp:
     val extension: ?loc:loc -> ?attrs:attrs -> extension -> expression
     val unreachable: ?loc:loc -> ?attrs:attrs -> unit -> expression
     
-    val case: pattern -> case_rhs -> case
+    val case: pattern -> guard:expression option -> expression -> case
```

Comment:
The original `ast_helper.mli` had:

```ocaml
val case : pattern -> ?guard:expression -> expression -> case
```

(See https://github.com/ocaml-flambda/flambda-backend/blame/1b6682660307f054113331959b7a064246a7a60b/ocaml/parsing/ast_helper.mli#L193.)

I'd prefer if this PR more consciously reverted the API to its state prior to your branch. (E.g. compiler libraries are included as part of the distribution, so user code can call functions like `Ast_helper.case`.)

Reply:
Done for this specific case. I will take a look at the overall diff to try to revert any APIs that don't need to be modified (and generally minimize the size of the diff).

Reply:
Did this!